astropy/table/typing.py

```diff
@@ -0,0 +1,15 @@
+from typing import Protocol
+
+from astropy.utils.data_info import MixinInfo
+
+__all__ = [
+    "MixinColumn",
+]
+
+
+class MixinColumn(Protocol):
+    info: MixinInfo
```

Comment:
I rather think the documentation is wrong (or at least misleading in how it uses the word "class"): concrete implementations *do* use an instance, rather than a class:
https://github.com/astropy/astropy/blob/a378c5e37698ad044e25bbebb4a38e275ec4ffc5/astropy/table/mixins/dask.py#L15

Reply:
The intent was to convey that `info` should be defined on the class rather than per instance. Probably the problem is `... which is a subclass ..` that should be `.. which is an instance of a subclass of ..`.

Reply:
should I open a separate PR to address it ?

Reply:
done in https://github.com/astropy/astropy/pull/17229